browser/brave_rewards/rewards_service_factory.cc

```diff
@@ -105,6 +101,11 @@ KeyedService* RewardsServiceFactory::BuildServiceInstanceFor(
   return rewards_service.release();
 }
 
+content::BrowserContext* RewardsServiceFactory::GetBrowserContextToUse(
+    content::BrowserContext* context) const {
+  return brave::IsRegularProfile(context) ? context : nullptr;
```

Comment:
this check should actually go in `BuildServiceInstanceFor`

Reply:
Done.